src/python/pybind11/core.cc

```diff
@@ -310,7 +310,7 @@ pybind_init_core(py::module_ &m_native)
         .def("disableAllListeners", &ListenSocket::disableAll)
         .def("listenersDisabled", &ListenSocket::allDisabled)
         .def("listenersLoopbackOnly", &ListenSocket::loopbackOnly)
-        .def("seedRandom", [](uint64_t seed) { random_mt.init(seed); })
+        //.def("seedRandom", [](uint64_t seed) { random_mt.init(seed); })
```

Comment:
This is a problem. As we don't have a single random generator anymore, we need to provide a way to seed all independent generators now. My suggestion is to keep the seedRandom (have a single seeder). Only it will store the seed in a global variable that will be used by the Random object (constructor) to init itself.

Reply:
I think I forgot to mention that I commented this for now in the PR but yes this sounded like an issue when I did it. Assuming I understood your suggestion, isn't there an assumption that seedRandom would be called before all Random objects are created ? Is that going to happen ?

If the goal is to have something callable from Python to reseed the random generator at any given time, we could keep a static (to the Random class) container of all existing Random objects and walk the container to reseed all of them on call to seedRandom.

Reply:
> I think I forgot to mention that I commented this for now in the PR but yes this sounded like an issue when I did it. Assuming I understood your suggestion, isn't there an assumption that seedRandom would be called before all Random objects are created ? Is that going to happen ?
> 

This is a good point. With SimObjects this assumption holds as they are dynamically created after m5.instantiate, but since the random generator can be used by any object we cannot rely on this assumption being true in all circumstances

> If the goal is to have something callable from Python to reseed the random generator at any given time, we could keep a static (to the Random class) container of all existing Random objects and walk the container to reseed all of them on call to seedRandom.

Yep, that's a good solution!

Reply:
@giactra Done, but I do not have a good idea of how to test it. Do you have a recommendation for an existing test that could call seedRandom?

Reply:
Good question. I believe you should take a simple config.

Then call

`m5.seedRandom(first_value)`

before m5.instantiate

Then call again 

`m5.seedRandom(second_value)`

And start simulation with m5.simulate

After simulation (m5.simulate), you should check for all Random objects to make sure they have the second seed. Now it looks like std::mt19937_64 does not have a getter to retrieve the seed (only a setter). As a workaround we could store the seed in the Random object after seeding has happened so we can retrieve it from there.

It's a bit of a hack and it requires adding debug only logic so let me know if you have better options in mind

Reply:
@giactra So I think ensuring all Random have the correct seed can be done with unittests rather than an actual simulation, unless I am missing some aspect.

I can add a static function of `Random`, let's call it `reseedAll(seed)`, that is what would be called from Python `seedRandom(seed)`. 

**Existing `Random` instances:** The expectation is that all `Random` objects will have the same seed after the call to `reseedAll(see)` , hence will produce the same first value, which we can hardcode and check against since `std::mt19937_64` is standard. The only quirk with this one is that several seeds may produce the same first random value (I don't know if this is possible with the Mersenne Twister RNG), so the test coverage would not be 100%. The alternative would be to have a debug only `seed` member in `Random`.

**Future `Random` instances:** The unit test could also check that after calling `reseedAll(seed)`, all future instances of Random are initialized with the seed passed by that call, by checking what number they provide first or a debug only member (as above).

What do you think ? This would also address @powerjg that we need some unittests for it, although those unittests would not look at interactions between components of which `Random` is a member (but I am not sure how to check this other than having your private or protected `Random` instance)